projects/plugins/jetpack/extensions/blocks/donations/amount/edit.js

```diff
@@ -73,23 +68,24 @@ const Amount = ( {
 		if ( isFocused || editedValue ) {
 			return;
 		}
-
 		setAmount( formatCurrency( defaultValue, currency, { symbol: '' } ) );
-	}, [ currency, defaultValue, editedValue, isFocused, setAmount ] );
+	}, [ currency, editedValue, isFocused, setAmount, defaultValue ] );
 
 	// Syncs the edited value with the actual value whenever the latter changes (e.g. new default amount after a currency change).
 	useEffect( () => {
 		if ( isFocused || isInvalid ) {
 			return;
 		}
-		setEditedValue( formatCurrency( value, currency, { symbol: '' } ) );
-	}, [ currency, isFocused, isInvalid, setAmount, value ] );
+		setEditedValue( formatCurrency( amount, currency, { symbol: '' } ) );
+	}, [ currency, isFocused, isInvalid, setAmount, amount ] );
 
 	return (
 		<div
-			className={ classnames( 'donations__amount', className, {
-				'has-focus': isFocused,
-				'has-error': isInvalid,
+			{ ...useBlockProps( {
+				className: classnames( 'wp-block-button wp-block-button__link', {
```

Comment:
For now we will rely on wp-block-button classes. We will use custom classes once we tackle [17230](https://github.com/Automattic/jetpack/issues/17230)